### DISCUSSION TITLE
Fresh sync

### DISCUSSION THREAD

**Discussion by addohm** [abraunegg/onedrive#1757] [Q&A] [answered]
How does one completely reset and start from scratch?  When I just delete everything and do a resync it only synced “changes”.  It looks like `onedrive —synchronize —resync` but the verbiage specifically says synching changes.  Is it just the verbiage or is it syncing the entire cloud drive from scratch?

**Comment by abraunegg** [comment, marked as answer]
@addohm 
> How does one completely reset and start from scratch? When I just delete everything and do a resync it only synced “changes”.
>..
> Is it just the verbiage or is it syncing the entire cloud drive from scratch?

OneDrive classifies files | folders | adds | removes | moves as a 'change'. This is why the application says it is syncing 'changes'.


> How does one completely reset and start from scratch?

1. Stop any active sync process or systemd service configuration
2. Logout of the application using `onedrive --logout`
3. Delete all files under `~/.config/onedrive`  or whatever your configuration directory is
4. Remove all data in ~/OneDrive locally
5. Re-authenticate the application by running `onedrive`
6. Sync your data via `onedrive --synchronize`
7. Enable any systemd service if applicable

**Note:** Only ever use `--resync` if the application tells you to post configuration change. Do not use it in normal sync operations.

**Comment by addohm** [comment]
@abraunegg thanks.  As a followup question, is there a way to verify the integrity of the downloaded onedrive content against what is online?  I'm trying to sync 700g and theres been several stoppages for one reason or another along the way.

**Comment by abraunegg** [comment]
@addohm 
>  As a followup question, is there a way to verify the integrity of the downloaded onedrive content against what is online?

There is no 'direct' way to show | verify the integrity without running the client in debug mode (`--verbose --verbose`). You can see these integrity checks occur when reviewing the debug log output. The integrity checks used are:
* File Hash as per provided by OneDrive
* File Size as data received as reported by OneDrive
* File Size as data written to disk as reported by OneDrive

If a file download (in normal download operations) fails these integrity checks it is marked as a bad file / failed download during the sync process.

**Reply by addohm** [reply]
Can I just repeatedly restart it by `onedrive --synchronize` until all the files appear to have been downloaded, then check their integrity as you'd mentioned?

**Reply by abraunegg** [reply]
> Can I just repeatedly restart it by onedrive --synchronize until all the files appear to have been downloaded, then check their integrity as you'd mentioned?

Not exactly. You can restart as many times as you like to ensure all the files are downloaded - just *dont* use `--resync` unless the application tells you to because you changed a configuration option that requires that action.

The integrity of the file is checked post the download - if the file downloaded successfully - the integrity is validated, at the point of file download & check completion is confirmed - thus, there is no further way to check outside the application. These integrity checks are *only* visible in debug log output mode.

To view these integrity checks, you must run the client in verbose debug mode (`--verbose --verbose`) and you can validate that log output post download - but that is going to be really onerous on you to perform going through the log file to manually validate your data.

Trust that the application is performing the integrity checks - and if the application marks a file as a bad download, that file failed its integrity check and is removed from local disk as a failed download.

If you are worried about local integrity of your files post download - use ZFS as your local filesystem format - nothing else.

**Comment by addohm** [comment]
Here is what I’ve been contending with.  I have close to 1TB on OneDrive.  I run a sync, and for reasons unknown it stops. I run a sync again, because there’s no apparent way to see if it’s completed.  At this point I’m starting to feel like I should just sync to a portable via the windows app and transfer the files to my NAS.

https://imgur.com/a/VMsekfK

**Comment by abraunegg** [comment]
@addohm 
> Here is what I’ve been contending with. I have close to 1TB on OneDrive. I run a sync, and for reasons unknown it stops.

The most common issue is poor Internet connectivity + this has been seen more often with Ubuntu systems due to how they run their Ubuntu Firewall.

If you are using Ubuntu, switch to a totally different Linux distribution - many folk who have encountered this, when they have switched distributions away from Ubuntu, the issue goes away.

To assist with this, review:
* https://github.com/abraunegg/onedrive/blob/master/docs/known-issues.md#application-stops-running-without-any-visible-reason
* https://github.com/abraunegg/onedrive/wiki/Debugging-SSL-TLS-Issues-when-the-client-stops-syncing-with-no-crash

> I run a sync again, because there’s no apparent way to see if it’s completed.

Yes there is - please review https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#all-available-commands as you have `--display-sync-status` to advise what your sync status is.

**Reply by addohm** [reply]
Switching to a different distro just because of this issue is really not a sensible suggestion, especially since it was the google search "ubuntu server onedrive sync" that got me here in the first place.  Given that I know MY connection isn't "flakey", that is to say, who knows down the pipe what could be happening, I think I will go down the road of syncing through windows and transferring the files over.  Thanks for all the help.  Hope the situation improves.
